### DISCUSSION TITLE
Welcome to openclassify Discussions!

### DISCUSSION THREAD

**Discussion by fatihalp** [openclassify/openclassify#858] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by topazstores** [comment]
I downloaded your code. How can I install on my server?

**Reply by fatihalp** [reply]
Have you checked this document? 

https://github.com/openclassify/openclassify#installation

**Reply by yableh** [reply]
Hello fatih,

I checked documentation but i can't get it installed on my cPanel hosting, Please could you give us a tutoriel on how to install this script on cPanel hosting ?

Thank you in advance.

**Comment by ShubhamPedgaonkar123** [comment]
How to create custom addons ?

**Reply by fatihalp** [reply]
Basically openclassify uses pyrocms module structure that you can just create module like here : 

https://pyrocms.com/videos/series/module-development

@ShubhamPedgaonkar123

**Comment by topazstores** [comment]
I followed the documentation but it still doesn't work. Can you assist?

**Reply by fatihalp** [reply]
its working for me now

**Reply by topazstores** [reply]
I managed to install. You can check my site https://sokonyeusi.co.ke . But I have one question, I wanted it to look like this openclassify demo https://beta.ocify.site/ because on my site, there is no search button on homepage

**Reply by fatihalp** [reply]
@topazstores it's all open source you can add this to header

```
<form  action="/advs/list" method="get">
<input  type="search" name="keyword"   placeholder="Enter keyword">
```

**Reply by topazstores** [reply]
Also, how can I make some ads to be featured? There's no such option in admin panel. I need users to be able to pay for their ads to be featured

**Reply by fatihalp** [reply]
it's advanced and premium module that you can buy at openclassify.com

**Comment by zkanoca** [comment]
I have  started installing but in the step 17 of 80 I get a MYSQL error:

17/80 Installing: Ads Module

In Connection.php line 712:

         
  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'json null' at line 1 (SQL
: alter table `co_advs_advs` add `foreign_currencies` json null)
         
In Exception.php line 18:
                                                                                                                                                                                                                         [3[14;201H 9;49m
  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'json null' at line 1  [3[14;201H
9;49m
                                                                                                                                                                                                                         [3[14;201H
9;49m

In PDOConnection.php line 82:
                                                                                                                                                                                                                         [3[14;201H
9;49m
  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'json null' at line 1  [3[14;201H
9;49m
                                                                                                                                                                                                                         [3[14;201H
9;49m

**Reply by zkanoca** [reply]
I use MariaDB. Since MariaDB use Longtext for JSON values I get that error.

**Reply by spektra2147** [reply]
Thank you for your feedback. Which version of MariaDb are you using?

**Reply by zkanoca** [reply]
I use MariaDB 10.1.9. It is OK, I switched to MySQL 8.0. It works now. 

Other issue is that I get error in another phase of the installation. 
ın the 40th step I get this error:  In RoleModel.php line 17:  Class "Anomaly\Streams\Platform\Model\Users\UsersRolesEntryModel" not found

**Reply by fatihalp** [reply]
I'm happy to hear that! @zkanoca :)

Can you tried with PHP 7.4 for now? We're working on 8.2

**Comment by zkanoca** [comment]
There is a bug in the installation process. I am asked to enter some information about server and database. Then the install fails after prompting. It tries to connect to database using with username "forge". Naturally, MySQL server refuses the connection request. Then I re-run the installation, now it connects and starts installation.

**Reply by spektra2147** [reply]
Thank you for your feedback.
This error occurs due to concurrent work during the creation of the .env file. I will deal with this issue.

**Reply by spektra2147** [reply]
https://github.com/openclassify/streams-platform/pull/43 @zkanoca

**Comment by adeelahmad5493** [comment]
Hi, Although it's kind of sounds dumb but still I am having the issue in finding the route which serves index page. 
Because '/' route is not defined anywhere or at least I couldn't find it but index page is being served on this endpoint. Can any one please help to find where is it mapped?

**Reply by fatihalp** [reply]
hi @adeelahmad5493 It is normal to have difficulty finding it because it is a modular system :)

Basically any module could have base route but normally pages module will manage routes. 

Also you can manage it inside theme like this one : https://github.com/openclassify/base-theme/blob/master/resources/views/layouts/default.twig#L30

**Reply by adeelahmad5493** [reply]
hi @fatihalp 
There are mainly two problems I need to resolve:
1. I basically wanted to style active tab while redirecting to different page. So, what I normally do is, storing active tab value in the session from controllers and in views I dynamically give class of activeTab to my home page tabs. That's why I wanted to know the controller and it's method serving index page.

2. I wanted to get some data on index page, which I usually pass while returning a view from controllers. 

Any hints, how could I achieve the desired functionality?

**Reply by adeelahmad5493** [reply]
Thanks @fatihalp 
I got it, it's mapped to pagesModule inside `storage/streams/default/pages/routes.php`.

**Comment by agenthinson99** [comment]
We offer all types of financial loans with an interest rate of 2%. Our offer ranges from a minimum amount of $10,000.00 USD up to a maximum amount of $10,000,000.00 USD for individuals & business purposes. We offer a wide range of financial services which includes: Business planning, commercial and development finance, properties and mortgages, debt consolidation loans, business loans, private loans, home refinancing, hotel loans, student loans etc. Kindly respond immediately via email below for more information.

Lender's Email: agenthinson@gmail.com

Interested applicants should apply with the following information below:-

Name:
Country:
Loan Amount: 
Duration:
WhatsApp Phone#:
Gender:
![Uploading aa.jpg…]()

**Comment by agenthinson99** [comment]
We offer all types of financial loans with an interest rate of 2%. Our offer ranges from a minimum amount of $10,000.00 USD up to a maximum amount of $10,000,000.00 USD for individuals & business purposes. We offer a wide range of financial services which includes: Business planning, commercial and development finance, properties and mortgages, debt consolidation loans, business loans, private loans, home refinancing, hotel loans, student loans etc. Kindly respond immediately via email below for more information.

Lender's Email: agenthinson@gmail.com

Interested applicants should apply with the following information below:-

Name:
Country:
Loan Amount: 
Duration:
WhatsApp Phone#:
Gender:
![aa](https://github.com/user-attachments/assets/6eb26214-2dbd-4931-8053-ab2a0db2fee0)
